packages/@lwc/engine-dom/src/custom-elements/create-pivot-constructor.ts

```diff
@@ -0,0 +1,462 @@
+const awaitingUpgrade = new Map<string, Set<HTMLElement>>();
+
+const EMPTY_SET = new Set();
+
+interface Definition {
+    UserCtor: CustomElementConstructor;
+    PivotCtor: CustomElementConstructor | undefined;
+    connectedCallback: (() => void) | undefined;
+    disconnectedCallback: (() => void) | undefined;
+    adoptedCallback: (() => void) | undefined;
+    attributeChangedCallback: ((name: string, oldValue: any, newValue: any) => void) | undefined;
+    observedAttributes: Set<string>;
+}
+
+function createDefinitionRecord(constructor: CustomElementConstructor): Definition {
+    const { connectedCallback, disconnectedCallback, adoptedCallback, attributeChangedCallback } =
+        constructor.prototype;
+    const observedAttributes = new Set(constructor.observedAttributes ?? []);
+    return {
+        UserCtor: constructor,
+        PivotCtor: undefined,
+        connectedCallback,
+        disconnectedCallback,
+        adoptedCallback,
+        attributeChangedCallback,
+        observedAttributes,
+    };
+}
+
+// Helper to create stand-in element for each tagName registered that delegates
+// out to the registry for the given element
+function createPivotingClass(tagName: string, registeredDefinition: Definition) {
+    class PivotCtor extends NativeHTMLElement {
+        constructor(UserCtor?: CustomElementConstructor) {
+            // This constructor can only be invoked by:
+            // a) the browser instantiating  an element from parsing or via document.createElement.
+            // b) LWC new PivotClass (This constructor is NOT observable/accessible in user-land).
+            // b) new UserClass.
+            // When LWC construct it, it will pass the upgrading definition as an argument
+            // If the caller signals via UserCtor that this is in fact a controlled
+            // definition, we use that one, otherwise fallback to the global
+            // internal registry.
+            super();
+            const definition = !isUndefined(UserCtor)
+                ? getOrCreateDefinitionForConstructor(UserCtor)
+                : globalDefinitionsByTag.get(tagName);
+            if (!isUndefined(definition)) {
+                internalUpgrade(this, registeredDefinition, definition);
+            } else {
+                // This is the case in which there is no global definition, and
+                // it is not handled by LWC (otherwise it will have a valid UserCtor)
+                // so we need to add it to the pending queue just in case it eventually
+                // gets defined in the global registry.
+                pendingRegistryForElement.set(this, registeredDefinition);
+                // We need to install the minimum HTMLElement prototype so that
+                // this instance works like a regular element without a registered
+                // definition; internalUpgrade will eventually install the full CE prototype
+                setPrototypeOf(this, HTMLElement.prototype);
```

Comment:
Looks like you're right! When I remove this line, the tests still pass. [The code originally came from the Polymer polyfill we forked](https://github.com/webcomponents/polyfills/blob/4756d228c16a0f53a739846edeaebd740bb9dc09/packages/scoped-custom-element-registry/src/scoped-custom-element-registry.js#L228-L231).

I'll remove it for now, but maybe @caridy has some ideas on why this might be required.